### PR TITLE
simplify VOTED status

### PR DESCRIPTION
**Pull request by liewegas**
- ALREADY_VOTED -> VOTED converge
- simplify UI to a drop-down that includes the VOTED :tada: status
- work around slack bug that prevented voter status block from updating

### CODE REVIEW

File: src/slack_interaction_handler.ts [hunk 1]
```diff
@@ -267,6 +262,24 @@ export async function handleVoterStatusUpdate({
         );
       }
 
+      // HACK: work around slack bug updating blocks: we must first remove the dropdown before we change it
+      const oldBlock = payload.message.blocks[2];
```
Comment (tomerovadia) [inline src/slack_interaction_handler.ts]: Is there a way to do this without hard-coding the 2? Maybe Andrew has a recommendation?

File: src/types.ts [hunk 2]
```diff
@@ -12,8 +12,7 @@ export type VoterStatus =
   | 'IN_PERSON'
```
Comment (tomerovadia) [inline src/types.ts:12]: What will happen if someone changes the dropdown of an old volunteer thread to ALREADY_VOTED (which will still be available in those dropdowns)?

File: src/slack_interaction_handler.ts [hunk 3]
```diff
@@ -267,6 +262,24 @@ export async function handleVoterStatusUpdate({
...
+      payload.message.blocks[2] = {
+        type: 'section',
+        block_id: 'asdf',
+        text: {
+          type: 'mrkdwn',
+          text: 'Updating...',
+          verbatim: false,
+        },
+      };
+      await SlackInteractionApiUtil.replaceSlackMessageBlocks({
```
Comment (tomerovadia) [inline src/slack_interaction_handler.ts]: I'm not sure I follow what's going on here. I know there's a bug but why are we calling replaceSlackMessageBlocks twice?

Comment (liewegas) [reply src/slack_interaction_handler.ts]: Updating the selected option doesn't work. So we first change the block to "Updating..." and then change it back to the drop down, which is enough to coax slack into selecting the new item.

File: src/types.ts (hunk 2 above)
Comment (liewegas) [reply src/types.ts:12]: ah, good catch! fixed this, but i had to rebase this branch on top of a merge of the others to fix conflicts.. so i'll rebase/repush once the others (follow-up, unclaim) merge first.

File: src/slack_interaction_handler.ts (hunk 3 above)
Comment (tomerovadia) [reply src/slack_interaction_handler.ts]: I see. Have you heard back from Andrew regarding this?

File: src/slack_interaction_handler.ts (hunk 1 above)
Comment (liewegas) [reply src/slack_interaction_handler.ts]: I don't think so.. we want to update the block in the third slot. That's hard-coded throughout.  We could use the replaceVoterPanelBlocks helper to make the change, but the oldBlocks = line is still needed.

This is a workaround for a bug that will hopefully get fixed someday (@fongandrew?) so I'm not too concerned about it.  It is moved into a helper in a later PR.

Comment (fongandrew) [reply src/slack_interaction_handler.ts]: I'm not sure I follow. What's the bug? The update isn't sticking?

Comment (fongandrew) [reply src/slack_interaction_handler.ts]: Oh, I see what it is. Yeah, it's on our radar and we'll fix this within the next week or two hopefully. In the meantime, I _think_ changing the `block_id` on the new block to something random will unstick it. Assuming that's what it is.

Comment (liewegas) [reply src/slack_interaction_handler.ts]: Yep, that worked! Thanks!

Comment (tomerovadia) [reply src/slack_interaction_handler.ts]: Thanks @fongandrew! @liewegas can you create an issue on this repo to remind us to fix this at some point in the future, and link to the issue from the comment explaining why this ID is necessary?

Comment (tomerovadia) [reply src/slack_interaction_handler.ts]: Regarding the hard-coding, I'm not seeing any other hardcoding like this in slack_interaction_handler -- are you seeing it throughout elsewhere? The only other place I'm seeing it is in slack_voter_status_message.

If you can create an issue in this repo to remind us to follow up on this another time I'm fine with submitting this as is, but hopefully we don't add too much more hard-coding.

**Review by tomerovadia — APPROVED** [review]

**Pull request merged**